tools/int_info/int_info.cc

```diff
@@ -0,0 +1,199 @@
+#include <RawPacket.h>
+#include <TcpLayer.h>
+#include <UdpLayer.h>
+#include <arpa/inet.h>
+#include <gflags/gflags.h>
+#include <sys/time.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "common/utils.h"
+#include "common/xnt.h"
+
+DEFINE_string(i, "", "The input pcap file.");
+DEFINE_string(o, "", "The file that stores output, default will be stdout.");
+
+int main(int argc, char* argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+  std::stringstream output;
+
+  if (FLAGS_i.empty()) {
+    output << "Input file name was not given" << std::endl;
+    return 1;
+  }
+
+  // open a pcap file for reading
+  pcpp::IFileReaderDevice* reader =
+      pcpp::IFileReaderDevice::getReader(FLAGS_i.c_str());
+
+  if (!reader->open()) {
+    delete reader;
+    output << "Error opening input pcap file\n";
+    return 1;
+  }
+
+  // print file summary
+  output << "File summary:" << std::endl;
+  output << "~~~~~~~~~~~~~" << std::endl;
+  output << "   File name: " << reader->getFileName() << std::endl;
+  output << "   File size: " << reader->getFileSize() << " bytes" << std::endl;
+
+  pcpp::PcapFileReaderDevice* pcap_reader =
+      dynamic_cast<pcpp::PcapFileReaderDevice*>(reader);
+
+  if (pcap_reader == nullptr) {
+    output << "Unknown file format";
+  }
+
+  uint32_t total_reports = 0;
+  uint32_t skipped = 0;
+
+  std::unordered_map<V4Tuple, uint64_t, V4TupleHasher> flow_reports; // <flow id, previous time>
+  std::vector<uint64_t> all_intervals;
+  std::unordered_set<uint32_t> flow_hashes;
+  pcpp::RawPacket raw_packet;
+  uint32_t prev_seq_no = 0;
+  size_t flow_with_multiple_report = 0;
+  while (pcap_reader->getNextPacket(raw_packet)) {
+    total_reports++;
+    pcpp::Packet parsedPacket(&raw_packet);
+
+    pcpp::UdpLayer* udp_layer = parsedPacket.getLayerOfType<pcpp::UdpLayer>();
+    if (!udp_layer) {
+      output << "No UDP header" << std::endl;
+      skipped++;
+      continue;
+    }
+    uint16_t l4_dport = ntohs(udp_layer->getUdpHeader()->portDst);
+
+    if (l4_dport != 32766) {
+      output << "UDP port not 32766: " << l4_dport << std::endl;
+      skipped++;
+      continue;
+    }
+    pcpp::PayloadLayer* payload_layer =
+        parsedPacket.getLayerOfType<pcpp::PayloadLayer>();
+
+    // Parse INT Header
+    uint8_t* payload = payload_layer->getPayload();
+    size_t payload_len = payload_layer->getPayloadLen();
+
+    std::shared_ptr<IntFixedHeader> int_fix_report =
+        ParseIntFixedHeader(&payload, &payload_len);
+
+    if (!int_fix_report) {
+      output << "No fix report" << std::endl;
+      skipped++;
+      continue;
+    }
+
+    uint32_t seq_no = ntohl(int_fix_report->seq_no);
+    if (prev_seq_no == 0) {
+      prev_seq_no = seq_no;
+    } else {
+      if (prev_seq_no != seq_no - 1) {
+        output << "Wrong sequence number: " << seq_no << ", should be "
```

Comment:
would be good to have a counter like `skipped`, to quickly identify how many "dropped" reports in the initial summary print